druid/src/widget/invalidation.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 The xi-editor Authors.
+//
```

Comment:
the way I might consider implementing this is not as a separate widget, but as part of `WidgetPod`'s `paint` method? It's possible this doesn't make sense, but it's my first instinct.

Reply:
I originally tried to imitate the implementation of `debug_paint_layout` (i.e. using `EnvScope` and `WidgetPod::paint` instead of a new widget). The problem is that, unlike drawing layout-rect boundaries, we don't want to recursively draw the invalidation regions for the children. It just occurred to me, though, that I can probably use `Painter` instead of a new widget.

Reply:
I see what you mean.